### PR TITLE
Experiment: OpenID Connect

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?

This PR adds the [OpenID Connect module]() to our installation profile, creates a Northstar OpenID provider, and configures it to pull user information from Northstar on login.
#### How should this be reviewed?

This is still all experimental, and shouldn't affect anything when the feature flag is off… although we do have the `/user/openid` route for _secret_ testing!
#### Any background context you want to provide?

References DoSomething/northstar#428.

💃 
#### Relevant tickets

Fixes DoSomething/northstar#423.
Fixes DoSomething/northstar#424.
Fixes DoSomething/northstar#425.
References DoSomething/northstar#426.
#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

**Comment by DFurnes** [comment]
Things we should continue with:
- [x] Add a feature flag for swapping out the login form. (@weerd)
- [x] Add a `/user/openid` route so we can test OpenID login even if toggled off. (@weerd)
- [ ] ~~Troubleshoot email conflict when authorizing as an existing Drupal account.~~ Migrate the custom `field_northstar_id` field to Drupal's `authmap` table so this module can find it.
- [ ] Add a registration page in Northstar. (@dfurnes)
- [ ] Persist the access token & refresh token in Phoenix between requests. (@dfurnes)
- [ ] Destroy the Northstar session when the user logs out of Phoenix.

**Comment by DFurnes** [comment]
So for the "email conflict when authorizing as an existing Drupal account" task above, it looks like it checks for existing Drupal users by comparing the `sub` claim (Northstar ID) against Drupal's `authmap` table – this makes total sense, but isn't how we're currently storing that value (it's in a custom field).

We'll need to make a little script to copy those values over, and then potentially drop the custom field.

File: lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+include_once('plugins/openid_connect_client/northstar/OpenIDConnectClientNorthstar.class.php');
+
+/**
+ * Register our custom OpenID Connect provider.
+ *
+ * Implements hook_ctools_plugin_directory().
+ */
+function dosomething_northstar_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'openid_connect') {
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc]: `===` triple equals?

**Comment by weerd** [comment]
¡OLÉ! :sheep: it!

**Review by weerd — COMMENTED** [review]
A minor comment on a rebellious ==

File: lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc (hunk 1 above)
Comment (DFurnes) [reply lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc]: 😱

**Comment by weerd** [comment]
¡OLÉ! :sheep: it!

**Pull request merged**